src/sdk/namespace_fs.js

```diff
@@ -612,6 +612,7 @@ class NamespaceFS {
             // TODO use file xattr to store md5_b64 xattr, etc.
             const stat = await nb_native().fs.stat(fs_account_config, upload_path);
             await nb_native().fs.rename(fs_account_config, upload_path, file_path);
+            await nb_native().fs.fsync(fs_account_config, path.dirname(upload_path));
```

Comment:
@jeniawhite Wait, isn't this the wrong path? we should be calling fsync on the target dir of the rename, which would be `path.dirname(file_path)` in this scope

Reply:
There's a link() call few lines above this line, that I think needs fsync too - and there it's indeed on the dirname of upload_path. Generally it seems that we want to follow any update to the filesystem with an fsync call to the directory of the updated file... that includes all these natiive fs calls - rename, link, unlink, mkdir, rmdir, writeFile, open('w'). We also said we want to have a control flag to turn off fsync.